File: contains-duplicate/contains-duplicate.js
// Brute force
// const containsDuplicate = function (nums) {
//     for (let i=0; i<nums.length;i++) {
//         for(let j=i+1;j<nums.length;j++) {
//             if(nums[i] === nums[j]) {
//                 return true;
//             }
//         }
//     }
//     return false;
// }

// optimal
const containsDuplicate = function (nums) {
    const seen = new Set()

    for (let num of nums) {
        if(seen.has(num)) {
            return true
        }
        seen.add(num);
    }

    return false;
}

// one liner
const oneLiner = function(nums) {
    return new Set(nums).size !== nums.length
}

const nums = [1,2,3,1]

console.log(containsDuplicate(nums))

File: contains-duplicate/contains-duplicate.py
# 217 - Contains Duplicate

#Brute force
# def containsDuplicate(nums):
#     for i in range(len(nums)):
#         for j in range(i+1,len(nums)):
#             if nums[i] == nums[j]:
#                 return True   
#     return False

# Optimal
def containsDuplicate(nums):
    seen = set()
    for num in nums:
        if num in seen:
            return True
        seen.add(num)
    return False

# One liner
# def containsDuplicate(nums):
#     return len(nums) != len(set(nums))

nums = [1,2,3,1]
print(containsDuplicate(nums))